### PR TITLE
Use try with resources, move unnecessary members to local vars

### PR DESCRIPTION
**Pull request by amcp**
Use try with resources
Make some class members final
Remove redundant throws declarations
Use JanusGraphFactory.drop() instead of .clear()
Inline some local variables and return
Remove unused methods
Make some members local
Check the result of File::mkdirs and File::delete

Addresses #783, #781, #740. #819, #820 in part

Signed-off-by: Alexander Patrikalakis <amcp@amazon.co.jp>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java [hunk 1]
```diff
@@ -97,13 +95,15 @@ private DataPuller addDataPuller(SliceQuery sq, StoreTransaction stx) throws Bac
...
+        final int numQueries;
         try {
             job.workerIterationStart(jobConfiguration, graphConfiguration, metrics);
 
             queries = job.getQueries();
             numQueries = queries.size();
-            Preconditions.checkArgument(numQueries>0,"Must at least specify one query for job: %s",job);
-            if (numQueries>1) {
+            Preconditions.checkArgument(numQueries >0,"Must at least specify one query for job: %s",job);
+            if (numQueries >1) {
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: add a space on the right side of the greater than sign.

File: janusgraph-core/src/main/java/org/janusgraph/util/datastructures/IterablesUtil.java [hunk 2]
```diff
@@ -47,16 +49,7 @@ public boolean apply(@Nullable E e) {
...
-
-            int count = 0;
-
-            @Override
-            public boolean apply(@Nullable O o) {
-                count++;
-                return count<=limit;
-            }
-        });
+        return StreamSupport.stream(iterable.spliterator(), false).limit(limit).collect(Collectors.toList());
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/util/datastructures/IterablesUtil.java]: undo this

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java [hunk 3]
```diff
@@ -112,12 +112,7 @@ static void foldInIds(final HasStepFolder janusgraphStep, final Traversal.Admin<
                 }
                 graphStep.addIds(ids);
                 if (!ids.isEmpty()) traversal.removeStep(currentStep);
-            }
-            else if (currentStep instanceof IdentityStep) {
-                // do nothing, has no impact
-            } else if (currentStep instanceof NoOpBarrierStep) {
-                // do nothing, has no impact
-            } else {
+            } else if (!(currentStep instanceof IdentityStep || currentStep instanceof NoOpBarrierStep)) {
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java]: make consistent with surrounding statements, !A && !B...

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/MixedConfiguration.java [hunk 4]
```diff
@@ -60,11 +60,10 @@ public boolean has(ConfigOption option, String... umbrellaElements) {
 
     @Override
     public Set<String> getContainedNamespaces(ConfigNamespace umbrella, String... umbrellaElements) {
-        Set<String> result = Sets.newHashSet();
-        for (ReadConfiguration config : new ReadConfiguration[]{global,local}) {
-            result.addAll(super.getContainedNamespaces(config,umbrella,umbrellaElements));
-        }
-        return result;
+        return Arrays.stream(new ReadConfiguration[]{global,local})
+            .map(config -> super.getContainedNamespaces(config,umbrella,umbrellaElements))
```
Comment (amcp) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/MixedConfiguration.java]: spaces after commas

File: janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java [hunk 5]
```diff
@@ -780,9 +780,7 @@ private Transaction(BaseTransactionConfig config) {
         private synchronized IndexSearcher getSearcher(String store) throws BackendException {
             IndexSearcher searcher = searchers.get(store);
             if (searcher == null) {
-                IndexReader reader = null;
-                try {
-                    reader = DirectoryReader.open(getStoreDirectory(store));
+                try (final IndexReader reader = DirectoryReader.open(getStoreDirectory(store))) {
```
Comment (amcp) [inline janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java]: this broke the lucene tests, undo

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/MixedConfiguration.java (hunk 4 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/MixedConfiguration.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java (hunk 1 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java (hunk 3 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/HasStepFolder.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/util/datastructures/IterablesUtil.java (hunk 2 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/util/datastructures/IterablesUtil.java]: ignore, this change was not at fault

File: janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEKeyValueStore.java [hunk 6]
```diff
@@ -146,40 +144,35 @@ public void acquireLock(StaticBuffer key, StaticBuffer expectedValue, StoreTrans
...
-            log.trace("db={}, op=getSlice, tx={}, resultcount={}", name, txh, result.size());
-//            log.trace("db={}, op=getSlice, tx={}, resultcount={}", name, txh, result.size(), new Throwable("getSlice trace"));
+        } catch (Exception e) {
+            throw new PermanentBackendException(e);
+        }
 
-            return new RecordIterator<KeyValueEntry>() {
-                private final Iterator<KeyValueEntry> entries = result.iterator();
+        log.trace("db={}, op=getSlice, tx={}, resultcount={}", name, txh, result.size());
+//      log.trace("db={}, op=getSlice, tx={}, resultcount={}", name, txh, result.size(), new Throwable("getSlice trace"));
```
Comment (davidclement90) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEKeyValueStore.java]: Dead code ?

Comment (amcp) [reply janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEKeyValueStore.java]: will remove and push.

**Pull request merged**